SU-HSDO/suhumsci
# STN-112 : Implement Table UI Pattern

# [STN-112](https://sparkbox.atlassian.net/browse/STN-112) READY FOR REVIEW

## Summary
Add implementation for the Table UI Pattern. This includes support for the "stacked cards" mobile behavior.

This PR also adds utility classes for pill styles. 
`.hb-pill-item` can be added to any text element to give it pill or badge styles
`.hb-pill-link` can be added to any text element which wraps an anchor tag
`.hb-pill-list` can be added to the `style settings` of any `field` item

![course tags style settings](https://user-images.githubusercontent.com/12678977/77369323-bb84f000-6d34-11ea-97e9-035e8bf8cbb9.png)

This PR has been reviewed by a designer (Merani) 👍
This PR has been browser tested and the following noted:
* JS functionality doesn't work as expected in IE due to a known JS bug where the `no-js` class is being applied to the `<html>` tag. See note in [JIRA STN-231](https://sparkbox.atlassian.net/browse/STN-231) regarding any JS imports that come after the decanter imports.

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. In the CLI run `npm run test` to verify all tests pass.
2. Import the [Duplicate of Custom Courses](https://drive.google.com/open?id=1z85g3DVtJr4fKifD1ad7oA0VxGtpHHrw) view
3. Test a 4 column table
    * Go to http://economics.suhumsci.loc/courses to see if you already have content. If so, check the view to confirm it is set to `Custom Courses`.
    * If you do not have content on the *Courses* page, add a `Component` with the `View` set to  `Custom Courses` and `Display` set to `All List`.
4. Test 5 column table
     * Create a `Basic Page`. Add a `Component` with the `View` set to `Duplicate of Custom Courses` and `Display` set to `All List`.
5. Test pill / badge utility classes
    * Add a `Text Area / WYSIWYG` component to a basic page. Add the following to the source code:
```
<p class="hb-pill-item">hb-pill-item</p>
<p><a class="hb-pill-link" href="http://hello.com">hb-pill-link</a></p>
```

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
